airbyte_cdk/sources/declarative/parsers/manifest_component_transformer.py

```diff
@@ -181,4 +208,34 @@ def propagate_types_and_parameters(
 
     @staticmethod
     def _is_json_schema_object(propagated_component: Mapping[str, Any]) -> bool:
-        return propagated_component.get("type") == "object"
+        return propagated_component.get("type") == "object" or propagated_component.get("type") == [
+            "null",
+            "object",
```

Comment:
this a a fix for a case when schema type is `["null", "object"]` and `$parameters` were not ignored and and were added to json schema